File: GymWare-Backend/GymWare.Entities/DietaCliente.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GymWare.Entities
{
    public class DietaCliente
    {
        public int DietaClienteId { get; set; }
        public virtual Dieta Dieta { get; set; }
        public virtual Cliente Cliente { get; set; }
        [Column(TypeName = "date")]
        public DateTime FechaInicio { get; set; }
        [Column(TypeName = "date")]
        public DateTime FechaFin { get; set; }
    }
}

File: GymWare-Frontend/src/app/classes/DietaCliente.ts
import { Dieta } from "src/app/classes/Dieta";
import { Cliente } from "src/app/classes/Cliente";

export class DietaCliente {

    DietaClienteId: number;
    Dieta: Dieta;
    Cliente: Cliente;
    FechaInicio: String;
    FechaFin: String;
}